malformed/test_chunking_example.py

```diff
@@ -0,0 +1,235 @@
+#
+# Three example tests are defined here:
+# 1. Test for forwarding the request with long and sophisticated header set.
+# 2. A simple request test for chunked response from the backend server.
+# 3. A simple request test transmitted over TLS.
+
+class CorrectHeadersTest(tester.TempestaTest):
+
+    backends = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'port' : '8000',
+            'response' : 'static',
+            'response_content' :
+"""HTTP/1.1 200 OK
+Content-Length: 0
+Connection: keep-alive
+
+"""
+        },
+    ]
+
+    tempesta = {
+        'config' : """
+cache 0;
+server ${general_ip}:8000;
+listen 80;
+listen 443 proto=https;
+tls_certificate ${general_workdir}/tempesta.crt;
+tls_certificate_key ${general_workdir}/tempesta.key;
+""",
+    }
+
+    clients = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80'
+        },
+        {
+            'id' : 'deproxy_ssl',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '443',
+            'ssl'  : True
+        },
+    ]
+
+    def iterate_test(self, test_func, request, *args, **kwargs):
+        # This function provides iterations over various chunk sizes.
+        # It have required positional parameters:
+        # @test_func - a function to iterate. Supposed, it implements
+        #              a test scenario and is a member of the same class.
+        # @request - a request to be sent
+        # The function can accept additional positional and keyword
+        # parameters with *args and **kwargs to forward them to
+        # the @test_func().
+        CHUNK_SIZES = [ 1, 2, 3, 4, 8, 16, 32, 64, 128, 256, 1500, 9216,
+                        1024*1024 ]
+        for i in range(len(CHUNK_SIZES)):
+            test_func(CHUNK_SIZES[i], request, *args, **kwargs)
+            if CHUNK_SIZES[i] > len(request):
+                break;
+
+    def inner_test_correct_headers(self, chunksize, request):
+        # This function defines a test scenario itself.
+        # @chunksize and @request are required positional parameters,
+        # to be in accordance with iterate_test().
+        #
+        # In this partucular example the test reproduses
+        # the fills_hdr_tbl_for_req() test from unit tests.
+        # However the fills_hdr_tbl_for_req() check for contents
+        # of internal TWF tables for parsed headers, whereas
+        # this test check for headers forwarded to the backend,
+        # so these two tests checks for different potential faults.
+        #
+        # In your case there could be your own scenario.
+        # And it can have additional positional or keyword
+        # parameters
+        deproxy_srv = self.get_server('deproxy')
+        deproxy_srv.start()
+        self.start_tempesta()
+        deproxy_cl = self.get_client('deproxy')
```

Comment:
Was it intentional to reload Tempesta FW for each new chunk size? Even with my recent speedup of tempesta loading detection (PR #199) the `start_tempesta()` still runs approx. 1 second, so the test takes a long time to complete.

Reply:
The truth is that TFW is not reloaded on every chunk size. Even the connection is not reestablished on every chunk size. I do not quite understand why this behaviour happens. However this is clearly visible with netconsole log and tcpdump.

Reply:
Basically we have https://github.com/tempesta-tech/tempesta-test/issues/113 to reduce Tempesta restarts over the whole framework, so let's postpone the issue until #113